src/api/test/functional/webui/user_controller_test.rb

```diff
@@ -181,6 +181,22 @@ def test_that_redirect_after_login_works # spec/controllers/webui/users_spec.rb
     assert_equal search_path, current_path
   end
 
+  def test_that_redirect_from_user_do_login_works # spec/controllers/webui/users_spec.rb
+    use_js
+
+    visit search_path
+    click_link("Log In")
+    fill_in 'Username', with: "tom"
+    fill_in 'Password', with: "brokenpassword"
+    click_button 'Log In'
+
+    fill_in 'Username', with: "tom"
+    fill_in 'Password', with: "buildservice"
+    click_button 'Log In'
+
+    assert_equal "tom", find('#home-username').text
```

Comment:
To also test the redirect you might want to add something like this:
assert_equal home_path, page.current_path